src/main/java/org/opensearch/flowframework/transport/GetWorkflowStateTransportAction.java

```diff
@@ -44,60 +48,96 @@ public class GetWorkflowStateTransportAction extends HandledTransportAction<GetW
-                        listener.onFailure(new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST));
-                    }
-                } else {
-                    listener.onFailure(new FlowFrameworkException("Fail to find workflow status of " + workflowId, RestStatus.NOT_FOUND));
-                }
-            }, e -> {
-                if (e instanceof IndexNotFoundException) {
-                    listener.onFailure(new FlowFrameworkException("Fail to find workflow status of " + workflowId, RestStatus.NOT_FOUND));
-                } else {
-                    String errorMessage = "Failed to get workflow status of: " + workflowId;
-                    logger.error(errorMessage, e);
-                    listener.onFailure(new FlowFrameworkException(errorMessage, RestStatus.NOT_FOUND));
-                }
-            }), context::restore));
+
+            resolveUserAndExecute(
+                user,
+                workflowId,
+                filterByEnabled,
+                listener,
+                () -> executeGetWorkflowStateRequest(request, listener, context),
+                client,
+                clusterService,
+                xContentRegistry
+            );
+
         } catch (Exception e) {
             String errorMessage = "Failed to get workflow: " + workflowId;
             logger.error(errorMessage, e);
             listener.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
         }
     }
+
+    /**
+     * Execute the get workflow state request
+     * @param request the get workflow state request
+     * @param listener the action listener
+     * @param context the thread context
+     */
+    private void executeGetWorkflowStateRequest(
+        GetWorkflowStateRequest request,
+        ActionListener<GetWorkflowStateResponse> listener,
+        ThreadContext.StoredContext context
+    ) {
+        String workflowId = request.getWorkflowId();
+        GetRequest getRequest = new GetRequest(WORKFLOW_STATE_INDEX).id(workflowId);
+        logger.info("Querying state workflow doc: {}", workflowId);
+        client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+            if (r != null && r.isExists()) {
+                try (XContentParser parser = ParseUtils.createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                    WorkflowState workflowState = WorkflowState.parse(parser);
+                    listener.onResponse(new GetWorkflowStateResponse(workflowState, request.getAll()));
+                } catch (Exception e) {
+                    String errorMessage = "Failed to parse workflowState: " + r.getId();
+                    logger.error(errorMessage, e);
+                    listener.onFailure(new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST));
+                }
```

Comment:
These aren't covered by tests.  Yeah, probably were missing coverage before you refactored, but we can add them now!

Reply:
Added couple of tests